models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiChatModel.java

```diff
@@ -195,29 +211,48 @@ public OpenAiChatModel(OpenAiApi openAiApi, OpenAiChatOptions options,
 	 * @param toolFunctionCallbacks The tool function callbacks.
 	 * @param retryTemplate The retry template.
 	 * @param observationRegistry The ObservationRegistry used for instrumentation.
+	 * @deprecated Use OpenAiChatModel.Builder or OpenAiChatModel(OpenAiApi,
+	 * OpenAiChatOptions, ToolCallingManager, RetryTemplate, ObservationRegistry).
 	 */
+	@Deprecated
 	public OpenAiChatModel(OpenAiApi openAiApi, OpenAiChatOptions options,
-			FunctionCallbackResolver functionCallbackResolver, List<FunctionCallback> toolFunctionCallbacks,
-			RetryTemplate retryTemplate, ObservationRegistry observationRegistry) {
-
-		super(functionCallbackResolver, options, toolFunctionCallbacks);
-
-		Assert.notNull(openAiApi, "OpenAiApi must not be null");
-		Assert.notNull(options, "Options must not be null");
-		Assert.notNull(retryTemplate, "RetryTemplate must not be null");
-		Assert.isTrue(CollectionUtils.isEmpty(options.getFunctionCallbacks()),
-				"The default function callbacks must be set via the toolFunctionCallbacks constructor parameter");
-		Assert.notNull(observationRegistry, "ObservationRegistry must not be null");
+			@Nullable FunctionCallbackResolver functionCallbackResolver,
+			@Nullable List<FunctionCallback> toolFunctionCallbacks, RetryTemplate retryTemplate,
+			ObservationRegistry observationRegistry) {
+		this(openAiApi, options,
+				LegacyToolCallingManager.builder()
+					.functionCallbackResolver(functionCallbackResolver)
+					.functionCallbacks(toolFunctionCallbacks)
+					.build(),
+				retryTemplate, observationRegistry);
+		logger.warn("This constructor is deprecated and will be removed in the next milestone. "
+				+ "Please use the OpenAiChatModel.Builder or the new constructor accepting ToolCallingManager instead.");
+	}
 
+	public OpenAiChatModel(OpenAiApi openAiApi, OpenAiChatOptions defaultOptions, ToolCallingManager toolCallingManager,
+			RetryTemplate retryTemplate, ObservationRegistry observationRegistry) {
+		// We do not pass the 'defaultOptions' to the AbstractToolSupport,
+		// because it modifies them. We are using ToolCallingManager instead,
+		// so we just pass empty options here.
+		super(null, OpenAiChatOptions.builder().build(), List.of());
```

Comment:
Do we still need to extend `AbstractToolCallSupport` class?

Reply:
We kept that to ensure API backward compatibility, but it will be removed right after M6 is released.